pkg/certsigner/signer.go

```diff
@@ -246,19 +306,34 @@ func (s *CertSigner) Sign(csr *capi.CertificateSigningRequest) (*capi.Certificat
 
 // getProfile returns the profile corresponding to the CSR Subject. For now only
 // `etcd-peers` and `etcd-servers` are considered valid profiles.
-func getProfile(csr *x509.CertificateRequest) (string, error) {
-	if csr.Subject.Organization != nil && len(csr.Subject.Organization) == 1 && csr.Subject.Organization[0] == "system:etcd-peers" {
-		if !strings.HasPrefix(csr.Subject.CommonName, "system:etcd-peer:") {
+func getProfile(csr *capi.CertificateSigningRequest) (string, error) {
+	x509CSR, err := csrutil.ParseCSR(csr)
+	if err != nil {
+		return "", fmt.Errorf("error parsing CSR, %v", err)
+	}
+	if err := x509CSR.CheckSignature(); err != nil {
+		return "", fmt.Errorf("error validating signature of CSR: %v", err)
+	}
+
+	if x509CSR.Subject.Organization != nil && len(x509CSR.Subject.Organization) == 1 && x509CSR.Subject.Organization[0] == "system:etcd-peers" {
```

Comment:
the 3 if blocks can be compacted as `x509CSR.Subject.Organization != nil && len(x509CSR.Subject.Organization) == 1` is common to all?

Reply:
yeah going to clean this up, was sloppy to not address.

Reply:
I think the update is a lot cleaner